File: Day3/169-Majority-Element/Method1.cpp
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        int maxi = 0, ans = 0;
        unordered_map<int, int> mpp;

        for (const auto it : nums)
        {
            mpp[it]++;
            if (mpp[it] > maxi)
            {
                maxi = mpp[it];
                ans = it;
            }
        }

        return ans;
    }
};

File: Day3/169-Majority-Element/Method1.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> hs = new HashMap<>();

        int maxi = 0, ans = 0;
        for (int num : nums) {
            hs.put(num, hs.getOrDefault(num, 0) + 1);
            int freq = hs.get(num);
            if (freq > maxi) {
                maxi = freq;
                ans = num;
            }
        }

        return ans;
    }
}